streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ProcessorNodeMetrics.java

```diff
@@ -289,6 +294,25 @@ public static Sensor processorAtSourceSensorOrForwardSensor(final String threadI
         return processAtSourceSensor(threadId, taskId, processorNodeId, streamsMetrics);
     }
 
+    public static Sensor e2ELatencySensor(final String threadId,
+                                          final String taskId,
+                                          final String processorNodeId,
+                                          final StreamsMetricsImpl streamsMetrics) {
+        final String sensorName = processorNodeId + "-" + RECORD_E2E_LATENCY;
+        final Sensor sensor = streamsMetrics.taskLevelSensor(threadId, taskId, sensorName, RecordingLevel.INFO);
```

Comment:
There's a slight hiccup with moving the INFO metrics from task to node level:
We get the current sensor from `StreamsMetrics#taskLevelSensor` which computes the `fullSensorName` with the `#taskSensorPrefix`
If we instead use `StreamsMetrics#nodeLevelSensor` then the` fullSensorName` is constructed from the `#nodeSensorPrefix`, which is obviously different. So moving this to a “true” node level sensor would be a breaking change, IIUC
I think the best we can do is just move this from TaskMetrics to ProcessorNodeMetrics, but still leave it as a taskLevelSensor. Let me know if I'm missing something though

Reply:
I do not think that using `StreamsMetrics#nodeLevelSensor()` is a breaking change. Sensors are not exposed publicly. They are merely containers for metrics. Metrics are exposed publicy. The full sensor name is just the key to retrieve sensors in the metrics map. It is also stored in `StreamsMetricsImpl` in the `*LevelSensor` data structures to know what sensors need to be removed from the metrics map when `removeAll*LevelSensors()` is called. So, the full sensor name is used for internal house keeping and hidden from the users. I think it is fine to change the `fullSensorName`.

Reply:
Oh, awesome, that's what I was hoping to hear! I'll move it to use the `nodeLevelSensor` then